docs/functional/architecture/0011-Scalar_vs_0d_Fields.md

```diff
@@ -0,0 +1,122 @@
+---
```

Comment:
- (+1) for keeping the scalar in the user interface

For the user's perspective it would be awkward having to write x:Field[[], float] if you want to just pass a float and I suspect they hardly ever really have a 0d Field. As a user, I would rather have the approach, I have a float, and if you internally treat that as a 0d Field, i don't care but don't bother me with it. 

- (+1) for keeping the type annotation the annotations of the function and not the annotated function

Anything else is somehow intransparent and confusing for the user. 

And I think it should be a major concern to keep a user friendly interface in the front end. So if it doesn not complicate that we do down the path, I think this is a good solution.